mlrun/api/crud/projects.py

```diff
@@ -72,7 +73,7 @@ def patch_project(
             session, name, project, patch_mode
         )
 
-    def delete_project(
+    async def delete_project(
```

Comment:
once you graduate a function to be async, you *must* follow all function inside calls to make sure all calls to IO are either async + await or sync and runs in threadpools. e.g. `verify_project_has_no_related_resources` must be run in threadpool.